vm/src/circuit/riscv.rs

```diff
@@ -417,7 +417,7 @@ fn parse_shamt(cs: &mut R1CS, Y: u32) {
 fn parse_J(cs: &mut R1CS, J: u32) {
     cs.set_var("J", J);
     for j in 1..=RV32::MAX_J {
-        cs.set_var(&format!("J={j}"), (j == J).into());
```

Comment:
This var is used elsewhere in the circuit (e.g., https://github.com/nexus-xyz/nexus-zkvm/blob/86e57135d0db805838dd9f42e77a1df63b9084a2/vm/src/circuit/riscv.rs#L589). Maybe we could add a test in `riscv.rs` that `set_var` and `set_bit` behave similarly with various arithmetic operations?

Reply:
It's mostly self-evident from the implementation
```
    pub fn set_bit(&mut self, name: &str, val: bool) -> usize {
        let j = self.new_var(name);
        self.w[j] = if val { ONE } else { ZERO };
        ...
```
and I can add a test about that.

Reply:
Yeah I'm just worried about an optimization causing a regression.